task/buildah-oci-ta/0.1/buildah-oci-ta.yaml

```diff
@@ -363,20 +363,23 @@ spec:
         [ -n "$COMMIT_SHA" ] && LABELS+=("--label" "vcs-ref=$COMMIT_SHA")
         [ -n "$IMAGE_EXPIRES_AFTER" ] && LABELS+=("--label" "quay.expires-after=$IMAGE_EXPIRES_AFTER")
 
+        ACTIVATION_KEY_PATH="/activation-key"
         ENTITLEMENT_PATH="/entitlement"
-        if [ -d "$ENTITLEMENT_PATH" ]; then
+
+        # do not enable activation key and entitlement at same time. If both vars are provided, prefer activation key.
+        # when activation keys are used, an empty directory on shared emptydir volume to /etc/pki/entitlement to prevent certificates from being included in the produced container.
```

Comment:
Do you specifically intend to make the methods mutually exclusive, or just to `ensure certificates generated by "subscription-manager register" are not included in the final build.`

I think we can achieve that ^ without making the mutually exclusive

Reply:
And if you do intend to make them mutually exclusive, should we prefer certs? IIUC activation keys require extra handling in the Containerfile, while the certs "just work"

Reply:
I think they should be made mutually exclusive and that is what I did here intentionally. I'd like to document that the the etc-pki-entitlement method of storing certs is discouraged and prone to random certificate revocation. Including activation keys now intentionally causes the certs generated by that method to be preferred.

Reply:
updated the PR with more info about my intentions here.

Reply:
Since the intended direction is activation keys, I think it is fine for that to be the overriding path even if it is more work than certs. The workaround is to remove the configuration for the unused method which should be a best practice anyway.